backend/core/illustration_service.py

```
@@ -0,0 +1,340 @@
+                file_size = abs_path.stat().st_size
+                max_size = 10 * 1024 * 1024  # 10MB limit
+                if file_size > max_size:
+                    logger.error(f"File too large: {file_size} bytes (max: {max_size})")
+                    raise ValueError(f"File too large: {file_size} bytes")
+
+            logger.info(f"Path validated successfully: {abs_path}")
+            return str(abs_path)
+
+        except (OSError, ValueError) as e:
+            logger.error(f"Path validation failed for '{path}': {e}")
+            raise ValueError(f"Invalid file path: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)